python/singa/autograd.py

```diff
@@ -924,17 +924,14 @@ def __init__(
         )
         self.W.gaussian(0.0, std)
 
-        if self.bias:
-            b_shape = (self.out_channels,)
-            self.b = Tensor(shape=b_shape, requires_grad=True, stores_grad=True)
-            self.b.set_value(0.0)
-        else:
-            # to keep consistency when to do forward.
-            self.b = None
-            # Tensor(data=CTensor([]), requires_grad=False, stores_grad=False)
+        b_shape = (self.out_channels,)
+        # forward method required bias variable to proceed (dummy b tensor)
+        self.b = Tensor(shape=b_shape, requires_grad=self.bias, stores_grad=self.bias)
```

Comment:
maybe use an empty tensor like this https://github.com/apache/incubator-singa/blob/0aae4c12cbc94d88a6c77793a7106c292925aac7/python/singa/autograd.py#L801

Reply:
It also has some updates on batchnorm operation, https://github.com/apache/incubator-singa/pull/468/files#diff-a236c495700ad62b39e28a4ece74711fR1095